app/presenters/ht_registration_presenter.rb

```diff
@@ -52,13 +63,33 @@ def show_dsp_name
     action == :index ? link_to(dsp_name, ht_registration_path(id)) : dsp_name
   end
 
-  def show_env
-    return unless self[:env].present?
+  def show_detail_display_name
+    env["HTTP_X_SHIB_DISPLAYNAME"]
+  end
+
+  def show_detail_edu_person_principal_name
+    env["HTTP_X_SHIB_EDUPERSONPRINCIPALNAME"]
+  end
 
-    fields = JSON.parse self[:env]
-    return unless fields.any?
+  def show_detail_geoip
+    geoip = Services.geoip.city ip_address
+    [geoip.country.name, geoip.most_specific_subdivision.name, geoip.city.name].join(" — ")
+  rescue => _e
+    ""
+  end
+
+  def show_detail_organizations
+    Services.whois.lookup(ip_address).parser.contacts
+      .map { |c| (c.name || c.organization || "").strip.gsub("\n", " — ") }
+      .join("<br/>")
+  rescue => _e
+    ""
+  end
 
-    ["<pre>", fields.map { |k, v| "<strong>#{k}</strong> #{v}" }, "</pre>"].join "\n"
+  def show_detail_reverse_lookup
+    Resolv.getname ip_address
```

Comment:
Doesn't seem to work in test, and we're swallowing the exception so it's not clear why.

Reply:
Should be addressed in the next commit with safety `require "resolv"` and some error handling.